fix: remove hard dependency on 'commons.compress'

jjohannes opened the pull request:
**Description**:

Follow up to #11656.

The removed statement added 'commons.compress' to the runtime **and compile** classpath of ALL modules. That's not needed and conceptually wrong. To update the version, the other changes of #11656 are sufficient.

**Related issue(s)**:

#11656

**Notes for reviewer**:
<!-- Provide logs, performance numbers or screenshots of the new functionality -->

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

rbarkerSL reviewed (approved):
LGTM

jjohannes commented:
I will check what's going on.

rbarkerSL commented:
Here's the gradle scan:

https://scans.gradle.com/s/zlmcizqwtvjza/console-log/task/:swirlds-logging:compileTestJava?anchor=13&page=1

On Mon, Jul 15, 2024 at 10:25 AM Jendrik Johannes ***@***.***>
wrote:

> I will check what's going on.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/hashgraph/hedera-services/pull/14240#issuecomment-2228772348>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BHKATRWSGR4WJPZIRRKOMS3ZMPSWVAVCNFSM6AAAAABK4TRGVOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDEMRYG43TEMZUHA>
> .
> You are receiving this because your review was requested.Message ID:
> ***@***.***>
>


-- 


Roger Barker

*Senior Platform Release Engineer*

***@***.***

SwirldsLabs.com <https://swirldslabs.com/>

jjohannes commented:
It is also failing locally for me. Apparently, this looked so trivial to me that I did not even test.

One thing I overlooked was that the dependency was defined as `api`, which means it was also added everywhere at **compile time** (and not only runtime). Which is event more reason to correct this, as we could produce code that compiles here but then does not run when used from Maven Central.

There was one place where a dependency was missing, which somehow the analysis did not catch. Fixed now.

rbarkerSL reviewed (approved):
LGTM
